bayes_opt/logger.py

```diff
@@ -117,14 +212,25 @@ def _is_new_max(self, instance):
         return instance.max["target"] > self._previous_max
 
     def update(self, event, instance):
+        """Handle incoming events.
+
+        Parameters
+        ----------
+        event : str
```

Comment:
Is this a string or an `Events` object? By looking at the code it is comparing `event == Events.OPTIMIZATION_START`, which implies that it is an Events object.

Reply:
These `Events.OPTIMIZATION_START` etc. are properties of the events classed with assigned string values `optimization:start` etc. I thought about going for `Event` as type, but that is technically not correct. Maybe it would be good to define `Events` as a proper [Enum](https://docs.python.org/3/library/enum.html) but I wouldn't do that in this PR.